src/controllers/user/controller.ts

```diff
@@ -14,112 +20,137 @@ class UserController {
+        const user = new UserRepository();
+        await user.get({ id })
+            .then((data) => {
+                res.send({
+                    message: 'User Fetched successfully',
+                    data: data,
+                    status: 200
+                });
+            });
     }
 
-    get(req: Request, res: Response, next: NextFunction) {
+    public get = async (req: Request, res: Response, next: NextFunction) => {
         try {
-            console.log('Inside get method of User');
+            const user = await this.userRepository.findAll(req.body, {}, {});
+            if (!user) {
+                next({
+                    message: 'Can not Find user',
+                    error: 404
+                })
+            }
             res.send({
-                message: 'User fetched succefully',
-                data: [{
-                    name: 'user1',
-
-                },
-                {
-                    name: 'user2',
-                }]
+                message: 'user fetched successfully',
+                data: user,
+                status: 200,
             });
         } catch (err) {
-            console.log('Inside err', err);
+            next({
+                message: 'Error while Fetching User'
+            })
         }
     }
-    create(req: Request, res: Response, next: NextFunction) {
+    public create = async (req: Request, res: Response, next: NextFunction) => {
         try {
-            console.log('Inside post method of Trainee');
+            const user = await this.userRepository.create(req.body, req.headers.user);
+            if (!user) {
+                next({
+                    message: 'User Not Created',
+                    error: 404,
+                })
+            }
             res.send({
-                message: 'User created succefully',
-                data: [{
-                    name: 'user1',
-
-                },
-                {
-                    name: 'user2',
-                }]
+                message: 'user created successfully',
+                data: user,
+                status: 200,
             });
         } catch (err) {
-            console.log('Inside err', err);
+            next({
+                message: 'Error while Creating User'
+            })
         }
     }
-    update(req: Request, res: Response, next: NextFunction) {
+    public update = async (req: Request, res: Response, next: NextFunction) => {
         try {
-            console.log('Inside put method of Trainee');
-            res.send({
-                message: 'Trainee updated succefully',
-                data: [{
-                    name: 'user1',
-
-                },
-                {
-                    name: 'user2',
-                }]
-            });
+            const id = req.params.id;
+            if (!id) {
+                next({
+                    message: 'id is Required',
+                    error: 404,
+                })
```

Comment:
Unnecessary checks, as this has been already checked in the validation middleware.